### PR TITLE
add try/catch to `permitAndStake`, `permitAndStakeMore`

### LINKED ISSUE #80: Add try/catch to `permitAndStake`, `permitAndStakeMore`
https://github.com/cantinasec/review-uniswapfoundation/issues/10

### PR DESCRIPTION
**Pull request by wildmolasses**
resolves #80

### CODE REVIEW

**Review by alexkeating — DISMISSED** [review]
Actually, it also looks like there is a failing test

**Comment by wildmolasses** [comment]
Fixed the tests, let me know if you think they satisfy

File: src/UniStaker.sol [hunk 1]
```diff
@@ -383,7 +383,7 @@ contract UniStaker is INotifiableRewardReceiver, Multicall, EIP712, Nonces {
     Deposit storage deposit = deposits[_depositId];
     _revertIfNotDepositOwner(deposit, msg.sender);
 
-    STAKE_TOKEN.permit(msg.sender, address(this), _amount, _deadline, _v, _r, _s);
+    try STAKE_TOKEN.permit(msg.sender, address(this), _amount, _deadline, _v, _r, _s) {} catch {}
```
[the remark this answers is not in the document]
Comment (wildmolasses) [reply src/UniStaker.sol:386]: added tests with explicit permit revert but stake success

**Review by alexkeating — APPROVED** [review]

**Review by apbendi — APPROVED** [review]
Looks good to me!

**Pull request merged**